File: examples/CORS.js
GlowScript 1.0 CoffeeScript

scene.range = 1.5
scene.forward = vec(-1,-.5,-1)
box
    size:vec(2,1,1)
    texture: "https://s3.amazonaws.com/glowscript/textures/cors_test.jpg"

s = 'This illustrates the use of an image from another web site as a texture.\n'
s += 'This is an example of CORS, "Cross-Origin Resource Sharing".'
scene.caption.text(s)

display_instructions = () ->
    s1 = "\n\nIn GlowScript programs:\n\n"
    s2 = "    Rotate the camera by dragging with the right mouse button,\n        or hold down the Ctrl key and drag.\n\n"
    s3 = "    To zoom, drag with the left+right mouse buttons,\n         or hold down the Alt key and drag,\n         or use the mouse wheel."
    scene.caption.append(s1+s2+s3)

# Display text below the 3D graphics:
display_instructions()

File: GlowScriptOffline/Demos/CORS.py
from vpython import *
#GlowScript 2.7 VPython

scene.range = 1
scene.forward = vector(-1,-.5,-1)
box(texture="https://s3.amazonaws.com/glowscript/textures/flower_texture.jpg")

s = 'This illustrates the use of an image from another web site as a texture.\n'
s += 'This is an example of CORS, "Cross-Origin Resource Sharing".\n'
scene.caption = s

scene.append_to_caption("""
Right button drag or Ctrl-drag to rotate "camera" to view scene.
To zoom, drag with middle button or Alt/Option depressed, or use scroll wheel.
  On a two-button mouse, middle is left + right.
Shift-drag to pan left/right and up/down.
Touch screen: pinch/extend to zoom, swipe or two-finger rotate.""")